dispatcher.go

```diff
@@ -154,41 +242,49 @@ type multicastDispatcher[T any] struct {
 	streams set[*asyncEventStream[T]]
 }
 
-// AddEventHandler adds a new event handler method that is called whenever an event T is dispatched. A
-// unique HandlerID is returned that can be used to remove the handler.
+// AddEventHandler adds a new event handler method that is called whenever an event T is
+// dispatched. A unique HandlerID is returned that can be used to remove the handler.
 func (md *multicastDispatcher[T]) AddEventHandler(handler EventHandler[T]) HandlerID {
 	return md.AddFilteredEventHandler(handler, nil)
 }
 
-// AddFilteredEventHandler adds a new event handler method that is called whenever a dispatched event T
-// passes the provided condition. A unique HandlerID is returned that can be used to remove the handler.
+// AddFilteredEventHandler adds a new event handler method that is called whenever a
+// dispatched event T passes the provided condition. A unique HandlerID is returned that can
+// be used to remove the handler.
 func (md *multicastDispatcher[T]) AddFilteredEventHandler(handler EventHandler[T], condition EventCondition[T]) HandlerID {
 	handlerID := HandlerID(uuid.NewString())
 	stream := md.NewFilteredEventStream(condition)
 	onClose := make(chan struct{})
 
+	ch := make(chan struct{})
```

Comment:
It's a little thing, but maybe a WaitGroup is more appropriate here?

Reply:
I've used both, don't have any strong opinions. However, it does seem like for a single wait, `chan` is likely preferable given that `WaitGroup` is synonymous to something like `CountDownLatch` in Java or `CountdownEvent` in C#, where you have potentially many things to wait on. However, even in those languages, I've used the latch types for a single event wait.